core/cont/inc/TList.h

```diff
@@ -85,10 +85,15 @@ friend  class TListIter;
    void              AddLast(TObject *obj) override;
    virtual void      AddLast(TObject *obj, Option_t *opt);
    void              AddAt(TObject *obj, Int_t idx) override;
+   virtual void      AddAt(TObject *obj, Int_t idx, Option_t *opt);
```

Comment:
Rather than declaration this `virtual` we need to also update to `TSeqCollection.h`

Reply:
Thanks. I was just copy pasting the structure from lines 85-86. AddLast overrides from TSeqCollection, AddLast with option is virtual. Otherwise it's not consistent?

Reply:
And if we also moved AddLast and AddFirst to TSeqCollection to be consistent, that would break user-scripts deriving from that classes since they would become purely virtual (if one wants to keep consistent style).

Reply:
And if we made them non-purely-virtual, then users might start getting warnings such as warning: ‘virtual void ::AddAt(TObject*, Int_t, Option_t*)’ was hidden ?

Reply:
> Otherwise it's not consistent?

Right.  I missed that.

> start getting warnings such as warning: ‘virtual void ::AddAt(TObject*, Int_t, Option_t*)’ was hidden ?

To extent that is better than use that:
* Inherits from TList
* Does not write/override `AddAt(TObject*, Int_t, Option_t*`
* Somehow starts using it ...
* It would 'silently' fail (depending what the overload does - but for example it would/was the case for `THashList`)

Reply:
v3 version here: https://github.com/root-project/root/pull/18389